lib/realtime/vehicle.ex

```diff
@@ -199,31 +204,87 @@ defmodule Realtime.Vehicle do
     end
   end
 
-  @spec scheduled_timepoint_status([StopTime.t()], Util.Time.timestamp()) ::
-          timepoint_status() | nil
-  def scheduled_timepoint_status([], _now) do
+  @doc """
+  If a block isn't scheduled to have started yet:
+    the start of the first trip
+  If a block is scheduled to have finished:
+    the end of the last trip
+  If now is in the middle of a layover:
+    the end of the previous trip
+  If now is in the middle of a trip:
+    the next timpeoint in that trip
+  """
+  @spec scheduled_location(Block.t(), Util.Time.timestamp()) :: scheduled_location() | nil
+  def scheduled_location(nil, _now) do
     nil
   end
 
-  def scheduled_timepoint_status(stop_times_on_trip, now) do
-    trip_start = List.first(stop_times_on_trip).time
-    now_time_of_day = Util.Time.nearest_time_of_day_for_timestamp(now, trip_start)
-    timepoints = Enum.filter(stop_times_on_trip, &is_a_timepoint?/1)
+  def scheduled_location([], _now) do
+    nil
+  end
 
-    case Realtime.Helpers.find_and_previous(timepoints, fn timepoint ->
-           timepoint.time >= now_time_of_day
-         end) do
-      nil ->
-        nil
+  def scheduled_location(block, now) do
```

Comment:
Could you just fetch and call `now_fn` inside `scheduled_location` so you don't need to pass it in?

Reply:
When unit testing, it's easier to pass in the values to a pure function, rather than having to set the environment for every test that uses a different input.
E.g.
```ex
block = ...
now = 1_546_362_000 # 2019-01-01 12:00:00 EST
assert Vehicle.scheduled_location(block, now) == ...
```
(https://github.com/mbta/skate/pull/84/files#diff-efc218bd9fe2e859af974b97d7770845R442)

as opposed to
```ex
block = ...
now = ...
Application.put_env(:realtime, :block_fn, fn _ -> block end)
Application.put_env(:realtime, :now_fn, fn -> now end)
assert Vehicle.scheduled_location(block_id, service_id) == ...
```

I still have to do that when testing the higher level `from_vehicle_position_and_trip_update`, but I'm not testing all the edge cases there.